### PR TITLE
Add and update notes about implicit restore

### LINKED ISSUE #14302: "dotnet publish" does not provide sufficient guidance when there is a private Nuget feed
This is feedback about the documentation of the "**dotnet publish**" command which is available on this [site](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore21).

I have a simple CSPROJ which references a nuget package from a private feed. The **dotnet publish** command fails with the following error:

> C:\Program Files\dotnet\sdk\3.0.100-preview9-014004\NuGet.targets(123,5): error : Unable to load the service index for source https://pkgs.dev.azure.com/kiteedge/_packaging/apex/nuget/v3/index.json.
C:\Program Files\dotnet\sdk\3.0.100-preview9-014004\NuGet.targets(123,5): error :   Response status code does not indicate success: 401 (Unauthorized)
---

I get a hint that **NUGET** is expecting credentials for the private **NUGET** feed.  However, the documentation provides no assistance to solve this problem.

You could provide additional links at the bottom of the page which directs the developer to the right source.  We have to understand that Visual Studio development and build is not something that can be taken lightly any more. 

While, I appreciate that Microsoft is giving the developer the opportunity to provide feedback through Github, this cannot be a substitute for sloppy in-house testing. Microsoft should take the responsibility of providing sensible documentation - documentation which has been tried and tested. 

#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 211afdc7-c796-202d-239d-823f2a499dff
* Version Independent ID: 9439897e-dcb3-bc2e-5cae-35c1ca2e0681
* Content: [dotnet publish command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore21)
* Content Source: [docs/core/tools/dotnet-publish.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-publish.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #14302

### CODE REVIEW

File: docs/core/tools/dotnet-restore.md [hunk 1]
```diff
@@ -28,10 +28,23 @@ dotnet restore -h|--help
 
 The `dotnet restore` command uses NuGet to restore dependencies as well as project-specific tools that are specified in the project file. By default, the restoration of dependencies and tools are executed in parallel.
 
-To restore the dependencies, NuGet needs the feeds where the packages are located. Feeds are usually provided via the *nuget.config* configuration file. A default configuration file is provided when the .NET Core SDK is installed. You specify additional feeds by creating your own *nuget.config* file in the project directory. You can override the *nuget.config* feeds with the - `-s` option.
+### Specify feeds
+
+To restore the dependencies, NuGet needs the feeds where the packages are located. Feeds are usually provided via the *nuget.config* configuration file. A default configuration file is provided when the .NET Core SDK is installed. To specify additional feeds:
+
+- Create your own *nuget.config* file in the project directory. For more information, see [Common NuGet configurations](/nuget/consume-packages/configuring-nuget-behavior) and [nuget.config differences](#nugetconfig-differences) later in this article.
```
Comment (gewarren) [inline docs/core/tools/dotnet-restore.md:35]: Are these steps one or the other, or both? If the former, maybe make that explicit by saying something like "To specify additional feeds, do one of the following:"

File: docs/core/tools/dotnet-restore.md [hunk 2]
```diff
@@ -28,10 +28,23 @@ dotnet restore -h|--help
...
+- Use `dotnet nuget` commands such as [`dotnet nuget add source`](dotnet-nuget-add-source.md).
+
+You can override the *nuget.config* feeds with the - `-s` option.
```
Comment (gewarren) [inline docs/core/tools/dotnet-restore.md:38]: Should this be
```suggestion
You can override the *nuget.config* feeds with the `-s` option.
```

File: includes/dotnet-restore-note-options.md [hunk 3]
```diff
@@ -1,4 +1,7 @@
...
->
-> This command also supports the `dotnet restore` options when passed in the long form (for example, `--source`). Short form options, such as `-s`, are not supported.
+You don't have to run [`dotnet restore`](~/docs/core/tools/dotnet-restore.md) because it's run implicitly by all commands that require a restore to occur, such as `dotnet new`, `dotnet build`, `dotnet run`, `dotnet test`, `dotnet publish`, and `dotnet pack`. To disable implicit restore, use the `--no-restore` option.
+
+The `dotnet restore` command is still useful in certain scenarios where doing an explicit restore makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control the time at which the restore occurs.
+
+For information about how to manage NuGet feeds, see the [`dotnet restore` documentation](../docs/core/tools/dotnet-restore.md).
+
+This command also supports the `dotnet restore` options when passed in the long form (for example, `--source`). Short form options, such as `-s`, are not supported.
```
Comment (gewarren) [inline includes/dotnet-restore-note-options.md:7]:
```suggestion
This command supports the `dotnet restore` options when passed in the long form (for example, `--source`). Short form options, such as `-s`, are not supported.
```

File: includes/dotnet-restore-note-options.md [hunk 4]
```diff
@@ -1,4 +1,7 @@
...
->
-> This command also supports the `dotnet restore` options when passed in the long form (for example, `--source`). Short form options, such as `-s`, are not supported.
+You don't have to run [`dotnet restore`](~/docs/core/tools/dotnet-restore.md) because it's run implicitly by all commands that require a restore to occur, such as `dotnet new`, `dotnet build`, `dotnet run`, `dotnet test`, `dotnet publish`, and `dotnet pack`. To disable implicit restore, use the `--no-restore` option.
+
+The `dotnet restore` command is still useful in certain scenarios where doing an explicit restore makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control the time at which the restore occurs.
```
Comment (gewarren) [inline includes/dotnet-restore-note-options.md:3]:
```suggestion
The `dotnet restore` command is still useful in certain scenarios where explicitly restoring makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control when the restore occurs.
```

File: includes/dotnet-restore-note.md [hunk 5]
```diff
@@ -1,3 +1,5 @@
...
-> It's still a valid command in certain scenarios where doing an explicit restore makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control the time at which the restore occurs.
+You don't have to run [`dotnet restore`](~/docs/core/tools/dotnet-restore.md) because it's run implicitly by all commands that require a restore to occur, such as `dotnet new`, `dotnet build`, `dotnet run`, `dotnet test`, `dotnet publish`, and `dotnet pack`. To disable implicit restore, use the `--no-restore` option.
+
+The `dotnet restore` command is still useful in certain scenarios where doing an explicit restore makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control the time at which the restore occurs.
```
Comment (gewarren) [inline includes/dotnet-restore-note.md:3]:
```suggestion
The `dotnet restore` command is still useful in certain scenarios where explicitly restoring makes sense, such as [continuous integration builds in Azure DevOps Services](https://docs.microsoft.com/azure/devops/build-release/apps/aspnet/build-aspnet-core) or in build systems that need to explicitly control when the restore occurs.
```

**Review by gewarren — APPROVED** [review]
Looks good, left a couple suggestions.

**Comment by tdykstra** [comment]
@gewarren Thanks for reviewing!

**Pull request merged**